WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadStarter.kt

```diff
@@ -141,40 +133,41 @@ class UploadStarter @Inject constructor(
 
     /**
      * This is meant to be used by [checkConnectionAndUpload] only.
+     *
+     * The method needs to be synchronized from the following reasons. When the app comes to foreground both
```

Comment:
> When the app comes to foreground both `queueUploadFromAllSites` and `queueUploadFromSite` are invoked

Not sure if this is true. The `queueUploadFromSite` is only called when the `PostListMainViewModel` is started or when swiping to refresh. 🤔 They can still run in parallel. But that's harder to explain. 😄

Reply:
Great eye ;)! I think what we originally wanted to write was "When the device connects to the internet both queueUploadFromAllSites and queueUploadFromSite are invoked` -> `queueUploadFromAllSites` is invoked from `connectionStatus` observer and `queueUploadFromSite` is invoked from the `UploadWorker`. I copy pasted the comment without thinking about it 😞 .